### ISSUE TITLE
Shield: Modify Address formulas to support encryption on Address fields

### ISSUE THREAD

**Issue by kevinswiggum** [SalesforceFoundation/HEDAP#652]
Several formula fields in HEDA use functions that are not supported by Shield Platform Encryption. 

Contact.Work_Address__c needs to remove reference to LEN function, replaced with ISBLANK.
Should be:
```
if(
  ISPICKVAL(Primary_Address_Type__c,"Work"),
  if(ISBLANK(MailingStreet) , "" , MailingStreet & ", ") & 
  if(ISBLANK(MailingCity) , "" , MailingCity & ", ")& 
  if(ISBLANK(MailingState) , "" , MailingState & " ")& 
  if(ISBLANK(MailingPostalCode) , "" , MailingPostalCode) & 
  If(ISBLANK(MailingCountry ) , "" , ", " & MailingCountry)
,
if(ISPICKVAL(Secondary_Address_Type__c,"Work"),
  if(ISBLANK(OtherStreet) , "" , OtherStreet & ", ") & 
  if(ISBLANK(OtherCity) , "" , OtherCity & ", ")& 
  if(ISBLANK(OtherState) , "" , OtherState & " ")& 
  if(ISBLANK(OtherPostalCode) , "" , OtherPostalCode) & 
  If(ISBLANK(OtherCountry ) , "" , ", " & OtherCountry)
  ,"" 
)
)
```

Address.Formula_MailingAddress__c needs to remove reference to LEN function replaced by ISBLANK
Should be
```
MailingStreet__c &
BR() & 
IF(ISBLANK(MailingStreet2__c), "", MailingStreet2__c & BR())  & 
IF(ISBLANK(MailingCity__c), "", MailingCity__c & ", ") & 
MailingState__c & " " & 
MailingPostalCode__c 
& IF(ISBLANK(MailingCountry__c), "", BR() & MailingCountry__c)
```
Address.Formula_MailingStreetAddress__c needs to remove reference to LEN function replaced by ISBLANK. Also needs to remove reference to TRIM function.
Should be
```
MailingStreet__c &
BR() & 
IF(ISBLANK(MailingStreet2__c), "", MailingStreet2__c & BR())  & 
IF(ISBLANK(MailingCity__c), "", MailingCity__c & ", ") & 
MailingState__c & " " & 
MailingPostalCode__c 
& IF(ISBLANK(MailingCountry__c), "", BR() & MailingCountry__c)
```

Testing this change will involve turning on platform encryption in a scratch org and encrypting all address-related fields on Contact and hed__Address.

**Comment by kevinswiggum** [comment]
**lurch: add

**Comment by LurchTheButler** [comment]
Tracking <a href="https://foundation.my.salesforce.com/a2x80000007fOW7AAM" target="_blank">W-028992</a>

**Comment by mrbelvedere** [comment]
Included in beta release 1.67 (Beta 2)
